### PR TITLE
Add serializer options specs

### PR DESCRIPTION
**Pull request by vr4b4c**
### Aim
Add specs for `InfinumJsonApiSetup::JsonApi::SerializerOptions`

#### Solution
- add factory bot and faker
- add location label association to location model (in order to later test `?include` param)
- add serializer options specs

### CODE REVIEW

File: spec/requests/api/v1/serializer_options_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
+require 'cgi'
+require 'uri'
+
+describe 'Serializer options', type: :request do
+  it 'adds meta with pagination information' do # rubocop:disable RSpec/MultipleExpectations
```
Comment (uncoverd) [inline spec/requests/api/v1/serializer_options_spec.rb:5]: Can we be more precise here and name this `with default offset pagination information` since the key checks below assume offset pagination?

File: spec/requests/api/v1/serializer_options_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    get '/api/v1/locations', params: {}, headers: default_headers
+
+    expect(response).to have_http_status(:ok)
+    expect(json_response).to have_key('meta')
+    meta = json_response['meta']
+    expect(meta).to have_key('current_page')
```
Comment (uncoverd) [inline spec/requests/api/v1/serializer_options_spec.rb:11]: Maybe `expect(meta.keys).to include('current_page', ...)`? I'm not sure how the error output is with have_key tho, I haven't tried it before 😓

File: spec/requests/api/v1/serializer_options_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    expect(links).to have_key('last')
+  end
+
+  it 'adds client requested fields information to links' do
+    q = { fields: { locations: 'latitude' } }
+    get "/api/v1/locations?#{q.to_query}", params: {}, headers: default_headers
+
+    expect(response).to have_http_status(:ok)
+    self_link = json_response.dig('links', 'self')
+    expect(parse_query(self_link)).to have_key('fields[locations]')
```
Comment (uncoverd) [inline spec/requests/api/v1/serializer_options_spec.rb:36]: This is a bit hard to parse, could we do a regex compare instead? It seems like string comparison at a glance 🤔 Same comment for the assertion below

**Review by uncoverd — CHANGES_REQUESTED** [review]

File: spec/requests/api/v1/serializer_options_spec.rb (hunk 1 above)
Comment (vr4b4c) [reply spec/requests/api/v1/serializer_options_spec.rb:5]: I'm OK with the current title (If I'm not mistaken this is the only pagination strategy currently supported by the query builder). Once the query builder gets updated we can then revisit this spec and change the title accordingly

File: spec/requests/api/v1/serializer_options_spec.rb (hunk 2 above)
Comment (vr4b4c) [reply spec/requests/api/v1/serializer_options_spec.rb:11]: 🤘 looks good on my end
```
||   1) Serializer options adds meta with pagination information
||      Failure/Error:
||        expect(json_response['meta'].keys).to include(
||          'current_page', 'total_pages', 'total_count', 'nekaj'
||        )
||      
||        expected ["current_page", "total_pages", "total_count", "padding", "page_size", "max_page_size"] to include "nekaj"
```

File: spec/requests/api/v1/serializer_options_spec.rb (hunk 3 above)
Comment (vr4b4c) [reply spec/requests/api/v1/serializer_options_spec.rb:36]: In this case, regex compare wouldn't work out-of-the-box because `[`, `]` are URL encoded.

I've played around with method naming and came up with following
```ruby
    expect(query_params(json_response.dig('links', 'self')))
      .to include('fields[locations]' => ['latitude'])
```

What do you think?

Comment (djordjelacmanovic) [reply spec/requests/api/v1/serializer_options_spec.rb:36]: I had something like this in mind

```ruby
parsed_query = parse_query json_response.dig('links', 'self')
expect(parsed_query).to have_key('fields[locations]')
```

**Review by djordjelacmanovic — APPROVED** [review]

File: spec/requests/api/v1/serializer_options_spec.rb (hunk 3 above)
Comment (uncoverd) [reply spec/requests/api/v1/serializer_options_spec.rb]: If I rate these in the order of mental capacity required to parse them (descending order) I come up with the following

```ruby
self_link = json_response.dig('links', 'self')
expect(parse_query(self_link)).to have_key('fields[locations]')
```

> what is parse query, what is self link

```ruby
  expect(query_params(json_response.dig('links', 'self')))
      .to include('fields[locations]' => ['latitude'])
```
> what is query params, why is the return value a hash

```ruby
expect(json_response.dig('links', 'self')).to match('lectures?fields%5Bself%5D=something')
```

> checking JSON response element against a string regex and bob's your uncle

Option 2 seems like a good middle ground but I will defer the decision to you since this is a nitpick 👍

**Review by uncoverd — APPROVED** [review]
